lib/urlToName.js

```diff
@@ -15,7 +15,7 @@ function urlToName(url) {
 
   return url
     .replace('https://registry.yarnpkg.com/', '') // prevents having long directory names
-    .replace(/[@/:-]/g, '_') // replace @ and : and - characters with underscore
```

Comment:
is it possible to add a test-case that reproduces the issue?

I think this happens when the URL contains `%2f` for example which might mean the URL has to be url-decoded instead.